### PR TITLE
add more detailed learning outcomes to readme

### LINKED ISSUE #5: more detailed learning outcomes
It reckon it helps to be really clear about what we want students to understand and be able to do after completing a workshop.

The following suggestions are mostly based on the existing material in the workshop, but also the suggested addition of a database (see issue #6 ).

Understand:
+ the importance of making passwords non-recoverable from a database
+ the difference between encrypting and hashing
+ why a password should be hashed before being stored
+ which hashing algorithms should be used and why
+ what a salt is and how it helps to protect against lookup table and rainbow table attacks
+ how salts should be generated and used
+ how a cryptographic hash function minimises vulnerability to brute force and dictionary attacks
+ what bcrypt is doing under the hood i.e. key stretching, iteration count

Be able to implement the following:
+ generate a random salt (using bcrypt.js)
+ hash a password (using bcrypt.js)
+ store this in a database
+ retrieve the hashed password from the database
+ compare it (using bcrypt.js) to the user input

### PR DESCRIPTION
**Pull request by jsms90**
Detail what we expect students to know and be able to do after completing this workshop.

To now include:
+ _why_ we take these measures
+ types of attacks we're protecting against
+ more detail about technical concepts

fixes #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
-__Learning Outcomes__
 
-- Why not to store plaintext passwords.
-- What is hashing?
-- Inneffective hashing vs bcrypt hashing.
-- How to use 'bcryptjs' npm package.
+### Learning outcomes
+Understand:
++ the importance of making passwords non-recoverable from a database
++ the difference between encrypting and hashing
```
Comment (ronanyeah) [inline README.md]: this is in WS3

File: README.md [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
++ why a password should be hashed before being stored
++ which hashing algorithms should be used and why
++ what a salt is and how it helps to protect against lookup table and rainbow table attacks
```
Comment (ronanyeah) [inline README.md]: what a salt is and why it is useful

File: README.md [hunk 3]
```diff
@@ -1,10 +1,22 @@
...
++ how salts should be generated and used
++ how a cryptographic hash function minimises vulnerability to brute force and dictionary attacks
```
Comment (ronanyeah) [inline README.md]: is redundant from line 4, no need to be so specific

File: README.md [hunk 4]
```diff
@@ -1,10 +1,22 @@
...
++ what bcrypt is doing under the hood i.e. key stretching, iteration count
```
Comment (ronanyeah) [inline README.md]: 'how bcrypt works'

File: README.md [hunk 5]
```diff
@@ -1,10 +1,22 @@
...
+
+Be able to implement the following:
++ generate a random salt (using bcrypt.js)
++ hash a password (using bcrypt.js)
++ store this in a database
++ retrieve the hashed password from the database
```
Comment (ronanyeah) [inline README.md]: no databases

File: README.md [hunk 6]
```diff
@@ -1,10 +1,22 @@
 ## Workshop 1 - Password Management
-__Learning Outcomes__
 
-- Why not to store plaintext passwords.
-- What is hashing?
-- Inneffective hashing vs bcrypt hashing.
-- How to use 'bcryptjs' npm package.
+### Learning outcomes
+Understand:
++ the importance of making passwords non-recoverable from a database
```
Comment (ronanyeah) [inline README.md]: i prefer 'why to avoid storing plaintext passwords'. 'non-recoverable from a database' is a bit alien

**Review by ronanyeah — CHANGES_REQUESTED** [review]
:+1:

File: README.md (hunk 6 above)
Comment (sohilpandya) [reply README.md]: Agreed.

**Pull request merged**